core/src/main/scala/com/databricks/labs/remorph/parsers/intermediate/functions.scala

```diff
@@ -1616,9 +1627,11 @@ case class BoolAnd(left: Expression) extends Unary(left) with Fn {
 }
 
 /** collect_list(expr) - Collects and returns a list of non-unique elements. */
-case class CollectList(left: Expression) extends Unary(left) with Fn {
-  override def prettyName: String = "COLLECT_LIST"
+case class CollectList(expr: Expression, cond: Option[Expression]) extends Expression with Fn {
```

Comment:
Spark has no `cond` parameter. convert it in the optimizer rule.
```
@ExpressionDescription(
  usage = "_FUNC_(expr) - Collects and returns a list of non-unique elements.",
  examples = """
    Examples:
      > SELECT _FUNC_(col) FROM VALUES (1), (2), (1) AS tab(col);
       [1,2,1]
  """,
  note = """
    The function is non-deterministic because the order of collected results depends
    on the order of the rows which may be non-deterministic after a shuffle.
  """,
  group = "agg_funcs",
  since = "2.0.0")
case class CollectList(
    child: Expression,
    mutableAggBufferOffset: Int = 0,
    inputAggBufferOffset: Int = 0) extends Collect[mutable.ArrayBuffer[Any]] {
```

also - can we change test examples to use COLLECT_LIST, which is more common?

Reply:
https://docs.databricks.com/en/sql/language-manual/functions/collect_list.html
<img width="1494" alt="Screenshot 2024-08-13 at 09 56 45" src="https://github.com/user-attachments/assets/6a2a805c-e416-4a62-bb7e-a8f166f638bd">